### PR TITLE
Updated joint limits for joint_t to match our robot

### PR DESCRIPTION
**Pull request by geoffreychiou**
@gomezc please review.

### CODE REVIEW

File: motoman_mh5_support/urdf/mh5.urdf [hunk 1]
```diff
@@ -135,7 +140,12 @@
     <child link="link_u"/>
     <origin rpy="0 0 0" xyz="0 0 0.3100"/>
     <axis xyz="0 -1 0"/>
-    <limit effort="100" lower="-2.3736" upper="4.4506" velocity="2.96"/>
+    <!-- Old limits: lower="-2.3736" upper="4.4506" The new limits
+           below were measured from an MH5F.  They ensure that link_r
+           won't hit link_l even when joint_r is at 45 degrees.  The
+           old limits were converted directly from the MH5 data sheet,
+           which was clearly way off. -->
+    <limit effort="100" lower="-1.0122" upper="3.2218" velocity="2.96"/>
```
Comment (gomezc) [inline motoman_mh5_support/urdf/mh5.urdf:148]: I'll assume this change too was done previously and the urdf was just not pushed with this change?

**Review by gomezc — APPROVED** [review]
urdf matches xacro so I'm approving; also limit changes look good

File: motoman_mh5_support/urdf/mh5.urdf (hunk 1 above)
Comment (geoffreychiou) [reply motoman_mh5_support/urdf/mh5.urdf:148]: Yes

**Comment by shaun-edwards** [comment]
These changes couldn't be pushed upstream (to ROS-I) because they are specific to our robot?  Is this a limitation we want?  Is there another way to achieve custom joint limits?  Does the SRDF support them?

**Comment by geoffreychiou** [comment]
> These changes couldn't be pushed upstream (to ROS-I) because they are specific to our robot? Is this a limitation we want? Is there another way to achieve custom joint limits? Does the SRDF support them?

We think someone limited it in the hardware/robot controller to avoid twisting the air hoses too far (so yea they are specific to our robot). 

Our moveit_config had joint limits specified but it doesn't seem like they were enforced:

```
  robot_joint_t:
    min_position: -1.2
    max_position: 3.14
    has_velocity_limits: true
    max_velocity: 6.97
    has_acceleration_limits: false
    max_acceleration: 0
```

The SRDF doesn't seem like it has any way to enforce joint limits.

The other option would be to disable the lock on the robot (if there is one?) and we can try to enforce the limit via software. Either way I want the URDF to reflect the values the robot gives me.

Which would you prefer?

**Comment by geoffreychiou** [comment]
It looks like at the top of joint_limits.yaml, it says:

```
# Joint limits can be turned off with [has_velocity_limits, has_acceleration_limits]
```

Does that mean if we set those values to false, the joint limits will be enforced? This could be a much simpler fix than what we are proposing to do.

**Comment by shaun-edwards** [comment]
>Does that mean if we set those values to false, the joint limits will be enforced? 

I don't think this is true. 

Since the URDF is specific to our setup, we should probably move the urdf to some place else.

One could imagine a different application with different limits. 

It's up to you though. I'm inclined to accept this as is in order to make forward progress.

**Comment by geoffreychiou** [comment]
> Since the URDF is specific to our setup, we should probably move the urdf to some place else.

We are merging it into the mh5_stem_cell branch so it's the perfect place :smile:.

Let me run some tests on this, if you add @gomezc to merge users for this she can help with it tomorrow.

**Comment by shaun-edwards** [comment]
>if you add @gomezc to merge users 

Done

**Comment by gavanderhoorn** [comment]
@geoffreychiou wrote:
> Our moveit_config had joint limits specified but it doesn't seem like they were enforced:

for those limits to be taken into account you need to have `has_position_limits: true` in that block as well (similar to `has_velocity_limits`, etc). See [this doxygen section](http://docs.ros.org/jade/api/joint_limits_interface/html/c++/namespacejoint__limits__interface.html#a84198ffacd73c3275d636e38a2948758) (there is other documentation, but it's inconvenient for me to look that up now).

I don't see that in the MoveIt config files, so that's probably why these limits weren't respected (as absence of that key equates to it being set to `false`).

**Comment by shaun-edwards** [comment]
@geoffreychiou, friendly reminder.

**Pull request merged**